ctriface/iface.go

```diff
@@ -314,42 +314,70 @@ func getImageURL(image string) string {
 
 func (o *Orchestrator) getImage(ctx context.Context, imageName string) (*containerd.Image, error) {
 	image, found := o.cachedImages[imageName]
-	if !found {
-		var err error
-		log.Debug(fmt.Sprintf("Pulling image %s", imageName))
-
-		imageURL := getImageURL(imageName)
-		local, _ := isLocalDomain(imageURL)
-		if local {
-			// Pull local image using HTTP
-			resolver := docker.NewResolver(docker.ResolverOptions{
-				Client: http.DefaultClient,
-				Hosts: docker.ConfigureDefaultRegistries(
-					docker.WithPlainHTTP(docker.MatchAllHosts),
-				),
-			})
-			image, err = o.client.Pull(ctx, imageURL,
-				containerd.WithPullUnpack,
-				containerd.WithPullSnapshotter(o.snapshotter),
-				containerd.WithResolver(resolver),
-			)
-		} else {
-			// Pull remote image
-			image, err = o.client.Pull(ctx, imageURL,
-				containerd.WithPullUnpack,
-				containerd.WithPullSnapshotter(o.snapshotter),
-			)
-		}
+
+	imageUrl := getImageURL(imageName)
+	if !found || imageIsOutdated(image, imageUrl) {
```

Comment:
Can you check how image staleness is checked in `containerd`? Clone the `containerd` repo and have a look into ContainerCreate stage of the sandbox creation in code, as well as in the logs. Containerd is already running on the worker node and you can check the performance of ContainerCreate phase in containerd logs. It should be creating queue-proxy container. Look into Estella's writing to see how to find logs.